src/redux/actions/passwordResetActions.js

```diff
@@ -0,0 +1,59 @@
+import axios from 'axios';
+import {
+  PASSWORD_REQUEST_LOADING, PASSWORD_REQUEST_SUCCESS, PASSWORD_REQUEST_FAILED, INITIAL_STATE,
+} from './types/resetPasswordType';
+
+export const requestLoading = () => ({
+  type: PASSWORD_REQUEST_LOADING,
+  payload: {
+    loading: true,
+  },
+});
+
+export const success = message => ({
+  type: PASSWORD_REQUEST_SUCCESS,
+  payload: {
+    loading: false,
+    success: true,
+    message,
+  },
+});
+
+export const failed = message => ({
+  type: PASSWORD_REQUEST_FAILED,
+  payload: {
+    loading: false,
+    message,
+  },
+});
+
+export const initialState = () => ({
+  type: INITIAL_STATE,
+  payload: {
+    loading: false,
+    success: false,
+    message: '',
+  },
+});
+
+export const requestPasswordResetLink = email => async (dispatch) => {
+  try {
+    dispatch(requestLoading());
+    const result = await axios.post('https://mazus-ah-staging.herokuapp.com/api/v1/auth/forgotpassword', { email });
+    dispatch(success(result.data.auth.message));
+  } catch (err) {
+    dispatch(failed(err.response.data.errors.message));
+    setTimeout(() => { dispatch(initialState()); }, 2000);
```

Comment:
I see that you've tried to use `setTimeout` to clear the error message after some seconds. Can we instead make sure the error message in the store is cleared before making any request? This way, there will be no error message to display when a user clicks the submit button after an error occurs.

Eg:
```
export const requestLoading = () => ({
  type: PASSWORD_REQUEST_LOADING,
  payload: {
    loading: true,
    message: ' ',
  },
});
```
Makes sense? Try it out and see if it works

Reply:
Ok... I'll try it now.

Reply:
It works! I'm taking out the setTimeout method and initialState action.
Thanks!